File: Python/Maximum Subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        max_sum = min(nums)
        array_sum = 0
        for num in nums:
            if array_sum > 0:
                array_sum += num
            else:
                array_sum = num
            if array_sum > max_sum:
                max_sum = array_sum
        return max_sum

File: cpp/Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int max_sum = -INFINITY;
        int array_sum = 0;
        for (int num : nums) {
            if (array_sum > 0) {
                array_sum += num;
            } else {
                array_sum = num;
            }
            if (array_sum > max_sum)
                max_sum = array_sum;
        }
        return max_sum;
    }
};
